nixos/modules/services/networking/bitcoind.nix

```diff
@@ -0,0 +1,194 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.bitcoind;
+  pidFile = "${cfg.dataDir}/bitcoind.pid";
+  configFile = pkgs.writeText "bitcoin.conf" ''
+    ${optionalString cfg.testnet "testnet=1"}
+    ${optionalString (cfg.dbCache != null) "dbcache=${toString cfg.dbCache}"}
+    ${optionalString (cfg.prune != null) "prune=${toString cfg.prune}"}
+
+    # Connection options
+    ${optionalString (cfg.port != null) "port=${toString cfg.port}"}
+
+    # RPC server options
+    ${optionalString (cfg.rpc.port != null) "rpcport=${toString cfg.rpc.port}"}
+    ${concatMapStringsSep  "\n"
+      (rpcUser: "rpcauth=${rpcUser.name}:${rpcUser.passwordHMAC}")
+      (attrValues cfg.rpc.users)
+    }
+
+    # Extra config options (from bitcoind nixos service)
+    ${cfg.extraConfig}
+  '';
+  cmdlineOptions = escapeShellArgs [
+    "-conf=${configFile}"
```

Comment:
Would it be useful to add a configFile option instead of having to use the configFile created in this module? In my case I'd like to be able to start bitcoind with a writeable config.

Reply:
agreed, one-off reindexing is kind of annoying when it's in your declarative config

Reply:
@jonasnick @jb55 I'm a little confused. Does `cfg.extraConfig` do what you want? Or do you want something like a `cfg.overrideConfig` string that would discard the generated `configFile`?

Reply:
A `configFile` option is needed, which can have a default value of the auto-generated file. People can then override it with e.g. `"/etc/bitcoind.conf"` which allows them to change the config at runtime. `extraConfig` doesn't allow that because you'd still have to rebuild NixOS to apply changes.